File: books-management-frontend/src/services/reviewService.js
const API_BASE_URL = '/api/reviews'

export const getAllReviews = async () => {
  const response = await fetch(API_BASE_URL)
  if (!response.ok) {
    throw new Error('Erro ao buscar avaliações')
  }
  return response.json()
}

export const getReviewById = async (id) => {
  const response = await fetch(`${API_BASE_URL}/${id}`)
  if (!response.ok) {
    throw new Error('Erro ao buscar avaliação')
  }
  return response.json()
}

export const getReviewsByBookId = async (bookId) => {
  const response = await fetch(`${API_BASE_URL}/book/${bookId}`)
  if (!response.ok) {
    throw new Error('Erro ao buscar avaliações do livro')
  }
  return response.json()
}

export const getBookStats = async (bookId) => {
  const response = await fetch(`${API_BASE_URL}/book/${bookId}/stats`)
  if (!response.ok) {
    throw new Error('Erro ao buscar estatísticas do livro')
  }
  return response.json()
}

export const createReview = async (reviewData) => {
  const response = await fetch(API_BASE_URL, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(reviewData),
  })
  if (!response.ok) {
    const error = await response.json()
    throw new Error(error.message || 'Erro ao criar avaliação')
  }
  return response.json()
}

export const updateReview = async (id, reviewData) => {
  const response = await fetch(`${API_BASE_URL}/${id}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(reviewData),
  })
  if (!response.ok) {
    const error = await response.json()
    throw new Error(error.message || 'Erro ao atualizar avaliação')
  }
  return response.json()
}

export const deleteReview = async (id) => {
  const response = await fetch(`${API_BASE_URL}/${id}`, {
    method: 'DELETE',
  })
  if (!response.ok) {
    throw new Error('Erro ao deletar avaliação')
  }
}

File: reviews-service/src/main/java/al/infnet/edu/br/reviews/application/service/ReviewService.java
package al.infnet.edu.br.reviews.application.service;

import al.infnet.edu.br.reviews.application.dto.ReviewCreateDTO;
import al.infnet.edu.br.reviews.application.dto.ReviewResponseDTO;
import al.infnet.edu.br.reviews.application.dto.ReviewStatsDTO;
import al.infnet.edu.br.reviews.application.dto.ReviewUpdateDTO;

import java.util.List;

public interface ReviewService {
    
    ReviewResponseDTO createReview(ReviewCreateDTO dto);
    
    ReviewResponseDTO getReviewById(String id);
    
    List<ReviewResponseDTO> getAllReviews();
    
    List<ReviewResponseDTO> getReviewsByBookId(Long bookId);
    
    ReviewResponseDTO updateReview(String id, ReviewUpdateDTO dto);
    
    void deleteReview(String id);
    
    void deleteAllByBookId(Long bookId);
    
    ReviewStatsDTO getBookStats(Long bookId);
}
